lib/browser/api/dialog.ts

```diff
@@ -287,5 +287,5 @@ export function showCertificateTrustDialog (windowOrOptions: BrowserWindow | Cer
 
   if (typeof message !== 'string') throw new TypeError('message must be a string');
 
-  return dialogBinding.showCertificateTrustDialog(window, certificate, message);
+  return dialogBinding.showCertificateTrustDialog?.(window, certificate, message);
```

Comment:
I'd rather we did a platform check in this method and threw instead of silently failing on linux

Reply:
I believe that check is done here https://github.com/electron/electron/blob/master/shell/browser/api/electron_api_dialog.cc#L100-L102

I'm not all-together-very-familiar with this codebase, so I'd need a pointer on how you'd like the check to happen if you'd also like it to happen in TypeScript

Reply:
Ah for sure, so that check you link is what makes `showCertificateTrustDialog` potentially undefined.  I'm saying that with this change something that used to throw an error, will now silently fail.  We should something like:

```js
if (!dialogBinding.showCertificateTrustDialog) {
  throw new Error(`showCertificateTrustDialog is not supported on this ${process.platform}`);
}
```

Reply:
cool, thanks, I added in 6ee87365eab02a4ad5e61c5d68a553f4805a5762